xdsl/rewriting/composable_rewriting/immutable_ir/immutable_ir.py

```diff
@@ -242,7 +242,7 @@ def is_type_seq(list: Sequence[Any]) -> TypeGuard[Sequence[Attribute]]:
 
         if is_type_seq(args):
             block_args: Sequence[IBlockArg] = [
-                IBlockArg(type, IList([]), self, idx) for idx, type in enumerate(args)
```

Comment:
Same here, I feel that this shouldn't have to be changed.
My understanding is that `pyright` is now unhappy about empty lists. empty tuples are still fine, because their type is `tuple[]`. However, `list` requires an argument.

Reply:
exactly, the error was unknown type. It's a stupid error but the fix is sort of trivial, and marginally more efficient